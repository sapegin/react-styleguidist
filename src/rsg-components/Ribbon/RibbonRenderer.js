import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

export const styles = ({ color, fontSize, fontFamily }) => ({
	root: {
		position: 'fixed',
		top: 0,
		right: 0,
		width: '149px',
		height: '149px',
		zIndex: 999,
	},
	link: {
		fontFamily: fontFamily.base,
		position: 'relative',
		right: -37,
		top: -22,
		display: 'block',
		width: 190,
		padding: '5px 15px',
		textAlign: 'center',
		color: 'white',
		fontSize: fontSize.base,
		background: color.ribbonBackground,
		textDecoration: 'none',
		textShadow: '0 -1px 0 rgba(0,0,0,.15)',
		transformOrigin: '0 0',
		transform: 'rotate(45deg)',
	},
});

export function RibbonRenderer({ classes, url, text, color, background }) {
	return (
		<div className={classes.root}>
			<a href={url} className={classes.link} style={{ color, background }}>
				{text || 'Fork me on GitHub'}
			</a>
		</div>
	);
}

RibbonRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	url: PropTypes.string.isRequired,
	text: PropTypes.string,
	color: PropTypes.string,
	background: PropTypes.string,
};

export default Styled(styles)(RibbonRenderer);
