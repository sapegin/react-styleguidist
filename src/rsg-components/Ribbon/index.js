export { default } from './Ribbon.js';
