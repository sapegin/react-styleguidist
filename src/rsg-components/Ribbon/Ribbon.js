import React from 'react';
import PropTypes from 'prop-types';
import RibbonRenderer from 'rsg-components/Ribbon/RibbonRenderer';

export default function Ribbon({}, { config }) {
	return <RibbonRenderer {...config.ribbon} />;
}

Ribbon.contextTypes = {
	config: PropTypes.object,
};
