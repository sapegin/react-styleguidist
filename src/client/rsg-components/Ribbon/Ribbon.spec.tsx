import React from 'react';
import { render, Provider } from '../../test';
import Ribbon from './Ribbon';

const url = 'http://example.com/';
const text = 'Share the repo';

it('should render ribbon if the ribbon is present in the config', () => {
	const { getByRole } = render(
		<Provider config={{ ribbon: { url } }}>
			<Ribbon />
		</Provider>
	);

	expect(getByRole('link')).toHaveAttribute('href', url);
});

it('should render ribbon with custom text', () => {
	const { getByText } = render(
		<Provider
			config={{
				ribbon: {
					url,
					text,
				},
			}}
		>
			<Ribbon />
		</Provider>
	);

	expect(getByText(text)).toBeInTheDocument();
});

it('should not render anything if the ribbon is not present in the config', () => {
	const { queryByRole } = render(
		<Provider config={{}}>
			<Ribbon />
		</Provider>
	);

	expect(queryByRole('link')).not.toBeInTheDocument();
});
