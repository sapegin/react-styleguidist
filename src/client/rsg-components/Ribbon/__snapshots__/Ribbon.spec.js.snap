// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ribbon should render ribbon if the ribbon is present in the config 1`] = `
<Styled(Ribbon)
  url="foo.bar"
/>
`;

exports[`RibbonRenderer should render ribbon with a text 1`] = `
<footer
  className="root"
>
  <a
    className="link"
    href="http://example.com"
  >
    Share the repo
  </a>
</footer>
`;

exports[`RibbonRenderer should render ribbon with url 1`] = `
<footer
  className="root"
>
  <a
    className="link"
    href="http://example.com"
  >
    Fork me on GitHub
  </a>
</footer>
`;
